snapshot-labs/sx-monorepo
# fix: refresh proposal after casting an offchain vote

### Summary



Closes: #235
Also fix #101 partially

Fix an issue where your freshly vote is not taken into account when computing anything relative to the proposal's total score, because the proposal is no refetched after the vote, and still uses the old `total_score`.

This PR refresh the proposal's store, to get the new `total_score` after casting a vote, so any internal navigation after casting a vote use the updated proposal data.

This quick fix only resolve the issue with offchain spaces. We need a more complete solution to handle both offchain and onchain spaces, with onchain proposal refresh being triggered only after tx executed. Maybe an Event store

### How to test

1. Go to a proposal
2. Cast a vote
3. The RESULTS section in the sidebar should refresh and take the new vote into account
4. Go to the votes tab without refreshing the page
5. It should display your voting power percentage correctly
